main.go

```diff
@@ -566,6 +632,72 @@ func main() {
 		log.Fatalf("Failed to add Get Wifi Status characteristic to device service: %s", err)
 	}
 
+	// set wifi connection
+	setWifiChar, err := deviceService.NewChar(setWifiUUIDFragment)
+	if err != nil {
+		log.Fatalf("Failed to create set wifi characteristic: %s", err)
+	}
+
+	setWifiChar.Properties.Flags = []string{
+		gatt.FlagCharacteristicEncryptAuthenticatedWrite,
+	}
+
+	setWifiChar.OnWrite(func(c *service.Char, value []byte) (resp []byte, err error) {
+		defer func() {
+			if err != nil {
+				log.Printf("Error setting wifi connection: %s.", err)
+			}
+		}()
+
+		var s string
+		err = json.Unmarshal(value, &s)
+		if err != nil {
+			log.Printf("Error setting wifi: %s", err)
+			return
+		}
+
+		var req setWifiRequest
+		err = json.Unmarshal([]byte(s), &req)
+		if err != nil {
+			log.Printf("Error setting wifi: %s", err)
```

Comment:
```suggestion
			log.Printf("Error unmarshaling wi-fi payload: %s", err)
```
Just to be more specific.